ynput/ayon-shotgrid
# Reimplementation of Tray user logging

## Changelog Description
We are reintroducing the functionality for users to access the ShotGrid addon via the tray widget. Previously, we had transitioned to an environment variable approach; however, following careful consideration of user input, we have opted to introduce two additional login methods. Users now have the option to log in using their username and password, mirroring the original process. Additionally, a novel method has been implemented where users can log in solely with their username, subsequently connecting to the ShotGrid Script API through studio settings. This approach involves a dedicated script with tailored permissions, catering specifically to fundamental user publishing requirements.

## Additional info
This PR is also solving several bugs caused by transition from OpenPype to ayon-core. Versoins are integrated again.

## Testing notes
Yeah this is long one! There are three ways to test this PR and we need to test all of them.

### Preparation for testing
All I will be describing in Dev mode ON
1. Since server settings were upgraded you need to update addon at your `ayon-docker/addons` folder. In repository root create and copy addon package to your local `ayon-docker` (this is still old structure so you should not clone this repo to addons folder directly). Or do it with the ZIP file uploading if you want to waist your time 😄 , or docker with server is not running from local machine. 
```
python .\create_package.py --skip-zip --output [PATH TO YOUR REPO]\ayon-docker\addon
```
2. Restart your server and then configure your bundle to lead to your client folder. 
![image](https://github.com/ynput/ayon-shotgrid/assets/40640033/320e90fc-4b7a-4a71-93cc-aa255636f9e1)
3. go to addon settings `ayon+settings://shotgrid/shotgrid_server` and add your server. (you can also set env `HTTPS_PROXY`)
4. in **core** addon go to `ayon+settings://core/publish/ExtractReview/profiles/0/outputs`, find h264 preset and add **shotgridreview** to tags - this will upload publishing version to your ShotGrid server.
5. Then follow other steps which are related to particular user login methods

### Configuring Local storage
You will need to have configured your ShotGrid server > Site settings > File Management before the publishing.
Since default settings are enabled and set to `primary` you need to make sure this is also correctly configured on your server.
Read more about it [here ](https://help.autodesk.com/view/SGSUB/ENU/?guid=SG_Administrator_ar_data_management_ar_linking_local_files_html)

### Publishing test steps
1. Make sure your Local storage settings are in sync with ShotGrid File Management presets. Look above for more info. **Configuring Local storage**.
2. Then open any host or Traypublisher and publish some sequence of video with **review** activated
3. All plugins should work well and version should be published to SG as expected

- [x] default Env method
    1. in `ayon+settings://shotgrid/client_login/env/client_sg_script_key` write the api key of ShotGrid script 
    2. in `ayon+settings://shotgrid/client_login/env/client_sg_script_name` write the name of ShotGrid script name 
    3. Save settings and you are good to open launcher in dev mode
    4. Make sure that `AYON_SG_USERNAME` environment variable is set to your SG registered email.
    5. Just in case the environment variable is correctly registered under your console either echo it or reboot the terminal.
    6. Follow **Publishing test steps**
    
- [x] User logs with password via Tray
    1. In settings `ayon+settings://shotgrid/client_login/type` set it to **Via Tray App with password**
    2. Save settings and Reboot your ayon launcher.
    3. input dialogue window should popup asking you for user@name and password you are using on SG
    4. Confirm and the window should automatically close after it is successfully registered
    5. Follow **Publishing test steps**

- [x] User logs only with name via Tray. 
    1. in settings go to `ayon+settings://shotgrid/client_login/type` and set it to **Via Tray App with shared api key**
    2. Go to your ShotGrid server and make new Script api access point for User restricted permissions. 
    3. Copy generated key and paste it to settings `ayon+settings://shotgrid/client_login/tray_api_key/client_sg_script_key` 
    4. in settings `ayon+settings://shotgrid/client_login/tray_api_key/client_sg_script_name` add the name of SG script. 
    5. Save the settings and reboot the launcher. 
    6. Dialogue window should popup with only _user@name_ to be added. (If it is not happening, go to Tray ShotGrid menu and click on change)
    7. Confirm the change and if all is successfully registered then the window should close.
    8. Follow **Publishing test steps**
 

closes #54
